### PR TITLE
add option to use validation loss to calculate loss stats

### PR DESCRIPTION
**Pull request by hsin-c**
2/2 of the 2 PRs for the issue: https://github.com/nv-morpheus/Morpheus/issues/497
This PR checks the "DFP to optionally use validation loss stats for zscore calculation instead just training loss stats" checkbox.
Temporarily pointing to the branch (in my fork) in the first PR to get a clean diff. After getting approval, would merge the first one and then re-target this to nv-morpheus.

### CODE REVIEW

**Review by gbatmaz — COMMENTED** [review]
by "loss stats", Do we mean the std and mean of the losses that are calculated by running inference on the val set after the training is completed and to be used in inference z score calculation? Or is this purely a training change?

**Comment by hsin-c** [comment]
> by "loss stats", Do we mean the std and mean of the losses that are calculated by running inference on the val set after the training is completed and to be used in inference z score calculation? Or is this purely a training change?

Exactly. It has nothing to do with training. This PR adds the option to use a validation set to get the losses to calculate the mean and std for the z score calculation during inference.

**Review by gbatmaz — APPROVED** [review]
LGTM

**Review by mdemoret-nv — APPROVED** [review]
